src/compiler/checker.ts

```diff
@@ -20849,7 +20849,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isEmptyAnonymousObjectType(type: Type) {

```

Comment:
Note that this is different from `isEmptyObjectType` and none of those reuse the other. Perhaps with this change, some sharing could be introduced. I tried that at first and a lot of tests failed - but I didn't bother digging into this right now. 

I don't really know what's the intended use case of both. Some code comments here would be highly appreciated. I could add the. However, I'd need first to learn from the team about the different goals of those two.